### ISSUE TITLE
Throw error when publish to zero peers

### ISSUE THREAD

**Issue by dapplion** [ChainSafe/lodestar#3650]
**Is your feature request related to a problem? Please describe.**

This bug https://github.com/ChainSafe/lodestar/pull/3641 would have been easily seen if when publishing to zero peers it errors

**Describe the solution you'd like**

Error when publish to zero peers. This is blocked by https://github.com/libp2p/js-libp2p-interfaces/pull/152

**Describe alternatives you've considered**

N/A

**Comment by twoeths** [comment]
@wemeetagain the libp2p-interfaces PR was merged, however I don't see it's included in any releases, could you double check?

**Comment by wemeetagain** [comment]
We're waiting for the typescript release of libp2p to publish this
We should merge #3661 first, then once libp2p 0.37 is released, another PR, and then after that resolve this issue.

**Comment by dapplion** [comment]
Implemented in latest gossipsub version

```ts
    if (tosend.size === 0 && !this.opts.allowPublishToZeroPeers) {
      throw Error('PublishError.InsufficientPeers')
    }
```

https://github.com/ChainSafe/js-libp2p-gossipsub/blob/703ea647aaddbfc78f6d69bcd23c852a6db199a9/ts/index.ts#L1785
